src/ore/src/process.rs

```diff
@@ -42,14 +42,32 @@
 macro_rules! halt {
     ($($arg:expr),* $(,)?) => {{
         $crate::__private::tracing::warn!("halting process: {}", format!($($arg),*));
-        $crate::process::halt();
+        $crate::process::exit_thread_safe(2);
     }}
 }
 
-/// Helper for the `halt!` macro.
+/// Prints the given message and exits the process.
+///
+/// `exit!` forwards the provided arguments to the [`tracing::info`] macro, then
```

Comment:
Defaulting to `0` seems a little surprising tbh; preference for forcing the exit code to be present, I think! And then the macro gets simpler too, which is nice.

Reply:
will do!